### ISSUE TITLE
Documentation: WakeOnLan depdendency is not clear to users

### ISSUE THREAD

**Issue by zalphoid** [NixOS/nixpkgs#323956]
## Problem

Its not clear to users that `networking.interfaces.<name>.wakeOnLan` requires `systemd-networkd`

## Proposal

Relocate the current wakeOnLan configuration setting to a subset of `systemd.network` and provide an alternative that creates a oneshot per interface to enable the [policy](https://search.nixos.org/options?channel=24.05&from=0&size=50&sort=relevance&type=packages&query=networking.interfaces.%3Cname%3E.wakeOnLan.policy) the user specifies leveraging ethtool.

###### OR

1. Update the documentation to state that it leverages `systemd-networkd`, for example, `Whether to enable wol on this interface using systemd-networkd`.
2. Add a dependency on `systemd-networkd` and  warn users if they wakeOnLan without it.

## Checklist

- [ ] checked [latest Nixpkgs manual] \([source][nixpkgs-source]) and [latest NixOS manual] \([source][nixos-source])
- [ ] checked [open documentation issues] for possible duplicates
- [ ] checked [open documentation pull requests] for possible solutions

[latest Nixpkgs manual]: https://nixos.org/manual/nixpkgs/unstable/
[latest NixOS manual]: https://nixos.org/manual/nixos/unstable/
[nixpkgs-source]: https://github.com/NixOS/nixpkgs/tree/master/doc
[nixos-source]: https://github.com/NixOS/nixpkgs/tree/master/nixos/doc/manual
[open documentation issues]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+label%3A%229.needs%3A+documentation%22
[open documentation pull requests]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+is%3Apr+label%3A%228.has%3A+documentation%22%2C%226.topic%3A+documentation%22

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
> `networking.interfaces.<name>.wakeOnLan` requires `systemd-networkd`

What makes you believe that?

https://github.com/NixOS/nixpkgs/blob/6afb255d976f85f3359e4929abd6f5149c323a02/nixos/modules/tasks/network-interfaces.nix#L1437-L1445

**Comment by zalphoid** [comment]
WOL isn't working on my system and I don't have systemd-networkd enabled.

If not systemd-networkd what else reads /etc/systemd/network/40-eth0.link?

**Comment by nipsy** [comment]
What's weird is that the "normal" scripted networking in NixOS (without using systemd-networkd) was working fine with WoL until around last month.  As I reported in what is probably a duplicate issue of this one at this point ( https://github.com/NixOS/nixpkgs/issues/339082 ), this was working fine without also using systemd-networkd, but recently broke, at least on all of my affected systems.

**Comment by zalphoid** [comment]
That's interesting, it might be worth checking to see if there were any changes to how WOL works but for me this was working fine back around 21.05 and then all of a sudden stopped working. Other than finding that it seems it requires the system to be using 'systems.network' I've just been too lazy to implement a new solution. I might give that a go if I have time this winter break.

**Comment by Majiir** [comment]
No, it does not require `systemd-networkd` (at least not by design).

> If not `systemd-networkd` what else reads [`/etc/systemd/network/40-eth0.link`](https://github.com/NixOS/nixpkgs/blob/a781ff33ae258bbcfd4ed6e673860c3e923bf2cc/nixos/modules/tasks/network-interfaces.nix#L1438-L1445)?

The answer is in `systemd.link(5)`:

> A plain ini-style text file that encodes configuration for matching network devices, used by systemd-udevd(8) and in particular its net_setup_link builtin.

We have systemd-udevd even when systemd-networkd is not enabled. This is why the wake-on-LAN implementation is shared by the scripted and networkd backends.

If it isn't working, that's a bug we should track in https://github.com/NixOS/nixpkgs/issues/339082. Unless someone recently (and deliberately) removed scripted backend support for wake-on-LAN, this isn't a documentation problem and we can close this issue.

**Comment by nipsy** [comment]
It's definitely worth mentioning the systemd-networkd and systemd.link/systemd-udevd associations for the `networking.interfaces.<name>.wakeOnLan.enable` option so it appears in the description of the option search web interface at search.nixos.org.  I know that is at least being extracted from nixpkgs directly which means that knowledge will be directly in the repository at that point.

Probably worth mentioning in the user manual also.  But I have no idea at the moment how that is being maintained in relation to the nixpkgs repo itself.
